src/wallet/walletdb.h

```diff
@@ -43,8 +43,16 @@ class CKeyMetadata
 {
 public:
     static const int CURRENT_VERSION=1;
+    static const int VERSION_SUPPORT_FLAGS=2;
+    static const uint8_t KEY_ORIGIN_UNSET         = 0x0000;
+    static const uint8_t KEY_ORIGIN_UNKNOWN       = 0x0001;
+    static const uint8_t KEY_ORIGIN_IMPORTED      = 0x0002;
+    static const uint8_t KEY_ORIGIN_UNENC_WALLET  = 0x0004;
+    static const uint8_t KEY_ORIGIN_ENC_WALLET    = 0x0008;
```

Comment:
At this point, it is probably the right approach (to use bitfield). It would mean to be extendable because it is a serialized and database stored uint8. Combinations should be possible.
But i agree that the `rpcwallet.cpp` part should be better. Will take a look at it.